server/auth/types/basic/basic_auth.ts

```diff
@@ -111,21 +111,13 @@ export class BasicAuthentication extends AuthenticationType {
         request,
         this.coreSetup.http.basePath.serverBasePath
       );
-      if (this.config.auth.anonymous_auth_enabled) {
-        const redirectLocation = `${this.coreSetup.http.basePath.serverBasePath}${ANONYMOUS_AUTH_LOGIN}?${nextUrlParam}`;
```

Comment:
When will `ANONYMOUS_AUTH_LOGIN` be called if this line is removed?

Reply:
Oh nvm, I see this is contained in `handleUnauthedRequest`. Does that mean its expected to log in as anonymous after getting an authentication failure?

Reply:
I;ve reverted this change to keep it simple.